src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -85,6 +86,29 @@ glibc_info()
   return _pt;
 }
 
+static std::string machine_info()
+{
+#if defined(__aarch64__) || defined(__arm__) || defined(__mips__)
```

Comment:
I have never been a fan of compiler #if def of variable declarations (they always are difficult to support).  By chance, could you refactor it to be something that would not require doing this.  For example:


```
#if defined(__aarch64__) || defined(__arm__) || defined(__mips__)
  #define MACHINE_NODE_PATH = "/proc/device-tree/model";
#elif defined(__PPC64__)
  #define MACHINE_NODE_PATH = "/proc/device-tree/model-name";
  // /proc/device-tree/system-id may be 000000
  // /proc/device-tree/model may be 00000
#elif defined (__x86_64__)
  #define MACHINE_NODE_PATH = "/sys/devices/virtual/dmi/id/product_name";
#else
  #error "Unsupported platform"
#endif

static std::string machine_info()
{
  static const char node[] = MACHINE_NODE_PATH;
  std::string model("unknown");
  std::ifstream stream(node);
  if (stream.good()) {
    std::getline(stream, model);
    stream.close();
  }
  return model;
}
```

Doing so keeps the keep readable easy to support.

Reply:
Done!